### PR TITLE
shipit-workflow: Use decision task for in action context's `taskGroupId`.

### PR DESCRIPTION
**Pull request by tomprince**
The action specification[1] documents this as being set to the decision                                                                                                                                            
task id of the task that defined the actions.                                                                                                                                                                      
                                                                                                                                                                                                                   
There is also discussion in scriptworker docs[2] suggesting that it is set                                                                                                                                         
differently for release promotion tasks, so that the created tasks get                                                                                                                                             
created in a new graph. However, it turns out that release-promotion always                                                                                                                                        
creates tasks in a task group named after the action tasks[3][4][5].                                                                                                                                               
                                                                                                                                                                                                                   
[1] https://docs.taskcluster.net/docs/manual/using/actions/spec#json-e-context                                                                                                                                     
[2] https://github.com/mozilla-releng/scriptworker/blob/77bb2f2adfd08ddb22e6cec4d8bc377a58222e87/scriptworker/cot/verify.py#L1240-L1245                                                                            
[3] https://searchfox.org/mozilla-central/rev/f822a0b61631cbb38901569e69b4967176314aa8/taskcluster/taskgraph/actions/release_promotion.py#313                                                                      
[4] https://searchfox.org/mozilla-central/rev/f822a0b61631cbb38901569e69b4967176314aa8/taskcluster/taskgraph/decision.py#173                                                                                       
[5] https://searchfox.org/mozilla-central/rev/f822a0b61631cbb38901569e69b4967176314aa8/taskcluster/taskgraph/create.py#41-47

### CODE REVIEW

**Comment by rail** [comment]
Looks like we need to fix [render_action_task](https://github.com/mozilla-releng/services/blob/8ed11df4c7477220cd160bef1f336d2de8c3416e/src/shipit_workflow/shipit_workflow/tasks.py#L118-L120) to properly handle `action_task_id` passed from [abandon_release](https://github.com/mozilla-releng/services/blob/8ed11df4c7477220cd160bef1f336d2de8c3416e/src/shipit_workflow/shipit_workflow/api.py#L172-L173). Or just add it to `context` when we call that API.

**Comment by tomprince** [comment]
I think we can actually just drop the parameter. It became unused [here](https://github.com/mozilla-releng/services/commit/732a570e5726c3ab60c2f562306f91604e1d2364). I'll post a follow up PR with that change (since I think it deserves its own commit message).

**Comment by rail** [comment]
I have no memories of that change :smile:

**Comment by escapewindow** [comment]
I think we have to set `taskGroupId` until we patch .taskcluster.yml, or we'll die on json-e render. Once we fix that, I'm ok removing this; we should double-check CoT verification passes afterwards in staging.

**Comment by tomprince** [comment]
Further investigation lead me to https://docs.taskcluster.net/docs/manual/using/actions/spec#json-e-context which suggests that `taskGroupId` should be specified, but it should be the decision task id the actions came from. I'll update this to match.

**Comment by tomprince** [comment]
I've updated this to pass in the decision task id, to match the action spec.

File: src/shipit_workflow/shipit_workflow/models.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def generate_phases(self, partner_urls=None, github_token=None):
             action_task_input['previous_graph_ids'] = list(previous_graph_ids)
             action_task_input['release_promotion_flavor'] = phase['name']
             action_task_id, action_task, context = generate_action_task(
+                decision_task_id=self.decicion_task_id,
```
Comment (escapewindow) [inline src/shipit_workflow/shipit_workflow/models.py]:
```
self.decicion_task_id
         ^
    *decision ?
```

File: src/shipit_workflow/shipit_workflow/api.py [hunk 2]
```diff
@@ -158,6 +158,7 @@ def abandon_release(name):
         for phase in filter(lambda x: x.submitted, release.phases):
             actions = fetch_actions_json(phase.task_id)
             action_task_id, action_task, context = generate_action_task(
+                decision_task_id=phase.task_id,
```
Comment (escapewindow) [inline src/shipit_workflow/shipit_workflow/api.py]: Which is this? the original decision task?

Comment (tomprince) [reply src/shipit_workflow/shipit_workflow/api.py]: This is the task-id of the task where the action was defined. Which in this case is the action task.

**Comment by rail** [comment]
@tomprince should we merge this?

File: src/shipit_workflow/shipit_workflow/models.py (hunk 1 above)
Comment (tomprince) [reply src/shipit_workflow/shipit_workflow/models.py]: That is an existing typo in the parameter.

**Comment by tomprince** [comment]
I think this can be merged now, yes. It appears that I don't have permission.

**Pull request merged**